tests/unit-tests/gconstruct/test_transform.py

```diff
@@ -633,6 +634,79 @@ def test_classification_processor():
     assert_equal(ret[stats_info_key][1], vals)
     assert_equal(ret[stats_info_key][2], counts)
 
+
+@pytest.mark.parametrize("out_dtype", [None, np.float16])
+def test_bucket_transform(out_dtype):
+    bucket_range = [10, 30]
+    transform = BucketTransform("test", "test", 2,
+                 bucket_range=bucket_range, slide_window_size=0, out_dtype=out_dtype)
+    feats = np.array([1, 11, 21, 31])
+    bucket_feats = transform(feats)
+    if out_dtype is not None:
+        assert bucket_feats['test'].dtype == np.float16
+    else:
+        assert bucket_feats['test'].dtype == np.float32
+
+    feats_tar = np.array([[1, 0], [1, 0], [0, 1], [0, 1]], dtype=out_dtype)
+    assert_equal(bucket_feats['test'], feats_tar)
+
+    # Case for float number
+    bucket_range = [1.1, 2.1, 3.1]
```

Comment:
Why there are three numbers for a bucket range?

Reply:
Test if it is provided with bucket range containing more elements, if it will fetch the maximum and minimum value

Reply:
Already change that to two value, and add assert about that.